dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitLabSourceCodeRepo.java

```diff
@@ -185,7 +188,7 @@ public void updateReferenceType(String repositoryId, Version version) {
 
     @Override
     protected String getCommitID(String repositoryId, Version version) {
-        //TODO: optimize here for gitlab by returning actual sha1
+        //TODO: Commit ID is returned by an existing API call
```

Comment:
Without understanding the whole flow, why can't this return `version.getCommitId()`? If not, what is the TODO that you need to do?

Reply:
It could, but the only time this is called for GitLab is while doing tool refresh, which is not yet handled, so it would be null anyways.

This PR is just for workflows (to deal with refresh) so I didn't touch that part.

Reply:
I'll update the todo to make more sense.

Reply:
Actually I'll just implement it.